splash/lua_runner.py

```diff
@@ -102,6 +106,9 @@ def dispatch(self, cmd_id, *args):
             try:
                 args = args or None
 
+                if self._is_stopped:
+                    raise StopIteration
```

Comment:
I'm not 100% sure this works correctly, or at least the code is not clear: the exception is caught later, and BaseScriptRunner tries to return the result by calling `self.on_result(res)`, where `res` is `lua2python(self.result)`. This looks unnecessary, and it is kind of hard to grasp what happens if `on_result` is called earlier, or if `self.result` is guaranteed to be None or if it can contain something else.

What do you think about raising another exception, and handling it explicitly? Or maybe just add a bit of comments to the source code which explain why does it work.